### PR TITLE
FOUNDATIONS: Notifications - Send Csv Pending Approval NotificationAsync

### PR DESCRIPTION
**Pull request by cjdutoit**
closes #AB15703

### CODE REVIEW

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+            ClientNotificationException clientNotificationException = new ClientNotificationException(
+                message: "Client notification error occurred, contact support.",
+                innerException: dependencyValidationException,
+                data: dependencyValidationException.Data);
+
+            var expectedNotificationDependencyValidationException =
+                new NotificationDependencyValidationException(
+                message: "Notification dependency validation error occurred, fix errors and try again.",
+                innerException: clientNotificationException);
```
Comment (JParkesAD) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs:37]: These lines should be indented

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                broker.SendEmailAsync(
+                    It.IsAny<string>(),
+                    It.IsAny<string>(),
+                    It.IsAny<string>(),
+                    It.IsAny<Dictionary<string, dynamic>>()))
+                        .ThrowsAsync(dependencyValidationException);
+
+            NotificationService notificationService =
+                new NotificationService(
+                    notificationConfigurations: invalidNotificationConfigurations,
```
Comment (JParkesAD) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs:49]: Indentation

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            NotificationConfigurations invalidNotificationConfigurations = this.notificationConfigurations;
+
+            ServerNotificationException serverNotificationException = new ServerNotificationException(
+                message: "Server notification error occurred, contact support.",
+                innerException: dependencyException,
+                data: dependencyException.Data);
+
+            var expectedNotificationDependencyException =
+                new NotificationDependencyException(
+                message: "Notification dependency error occurred, contact support.",
```
Comment (JParkesAD) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs:98]: Indentation

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+            var failedServiceNotificationException =
+                new FailedServiceNotificationException(
+                    message: "Failed service notification error occurred, contact support.",
+                    innerException: someException,
+                    data: someException.Data);
+
+            var expectedNotificationServiceException =
+                new NotificationServiceException(
+                message: "Notification service error occurred, contact support.",
```
Comment (JParkesAD) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs:160]: Indentation

File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Server/Controllers/ReIdentificationController.cs [hunk 5]
```diff
@@ -115,7 +115,7 @@ public async ValueTask<ActionResult<AccessRequest>>
             }
         }
 
-        [HttpGet("{csvreidentification}")]
+        [HttpGet("{csvreIdentification}")]
```
Comment (JParkesAD) [inline ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Server/Controllers/ReIdentificationController.cs:118]: Should this be csvReIdentification?

File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Controllers/ReIdentificationController.cs [hunk 6]
```diff
@@ -115,7 +115,7 @@ public async ValueTask<ActionResult<AccessRequest>>
             }
         }
 
-        [HttpGet("{csvreidentification}")]
+        [HttpGet("{csvreIdentification}")]
```
Comment (JParkesAD) [inline ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Controllers/ReIdentificationController.cs:118]: Should this be csvReIdentification?

File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Server/Controllers/ReIdentificationController.cs (hunk 5 above)
Comment (cjdutoit) [reply ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Server/Controllers/ReIdentificationController.cs:118]: fixed, should be all lower case for route

File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Controllers/ReIdentificationController.cs (hunk 6 above)
Comment (cjdutoit) [reply ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server/Controllers/ReIdentificationController.cs]: fixed

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs (hunk 4 above)
Comment (cjdutoit) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs:156]: fixed

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs (hunk 1 above)
Comment (cjdutoit) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs:36]: fixed

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs (hunk 2 above)
Comment (cjdutoit) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs]: fixed

File: ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs (hunk 3 above)
Comment (cjdutoit) [reply ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/Notifications/NotificationTests.Exceptions.SendCsvPendingApprovalNotification.cs:95]: fixed

**Pull request merged**